juju/juju
# Adds upgrade worker to model agent

## Please provide the following details to expedite Pull Request review:

### Checklist


----

## Description of change

*Please replace with a description about why this change is needed, along with a description of what changed?*

## QA steps

*Please replace with how we can verify that the change works?*

```sh
cd tests && ./main.sh caasadmission
```

2nd
```sh
export DOCKER_USERNAME=<your-dockerhub-username>
make install operator-image microk8s-operator-update push-operator-image
juju bootstrap microk8s
# Check that the model agent has come up in the controller namespace
juju add-model test
# Check that the model agent has come up in the test namespace

export JUJU_BUILD_NUMBER=1
make install operator-image microk8s-operator-update push-operator-image
juju upgrade-controller --agent-stream=develop --agent-version=2.8.1.1
# Wait for controller to restart and verify that the new controller that comes up has the new image and version. The model agent has the correct version and image.
```

## Documentation changes

*Please replace with any notes about how it affects current user workflow? CLI? API?*

## Bug reference

*Please add a link to any bugs that this change is related to.*
